app/src/main/java/com/example/login/viewmodel/UserAuthorizationViewModel.kt

```diff
@@ -0,0 +1,71 @@
+package com.example.login.viewmodel
+
+import android.view.View
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.navigation.Navigation
+import com.example.login.R
+import com.example.login.model.UserModel
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+
+class UserAuthorizationViewModel: ViewModel() {
+     val userNameModel: MutableLiveData<String> by lazy {
+        MutableLiveData<String>()
+    }
+     val userSurnameModel: MutableLiveData<String> by lazy {
+        MutableLiveData<String>()
+    }
+     val userAgeModel: MutableLiveData<Int> by lazy {
+        MutableLiveData<Int>()
+    }
+
+    val correctAnswerModel: MutableLiveData<Int> by lazy {
+        MutableLiveData<Int>()
+    }
+    val unCorrectAnswerModel: MutableLiveData<Int> by lazy {
+        MutableLiveData<Int>()
+    }
+
+    private val statusMessage = MutableLiveData<Event<Int>>()
+    val message : LiveData<Event<Int>>
+        get() = statusMessage
+
+    fun confirmUserInfo(userModel: UserModel, view: View){
+        val allSymbols: Pattern = Pattern.compile("[^A-Za-z0-9-А-Я-а-я]")
+        val matcherByName: Matcher = allSymbols.matcher(userModel.name)
+        val matcherBySurname: Matcher = allSymbols.matcher(userModel.surname)
+        val nameHasSpecialSymbols: Boolean = matcherByName.find()
+        val surnameHasSpecialSymbols: Boolean = matcherBySurname.find()
+        val normal = true
+        when{
+            userModel.name.isEmpty() && userModel.surname.isEmpty() -> statusMessage.value = Event(R.string.two_lines_empty)
+
+            userModel.name.isNotEmpty() && userModel.surname.isEmpty() -> statusMessage.value = Event(R.string.need_add_surname)
```

Comment:
ViewModel should not referenced to R.string

Reply:
done